xgcm/test/test_padding.py

```diff
@@ -0,0 +1,1120 @@
+            y: (0, boundary_width["Y"][1]),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+
+    # Now pad each face according to the swapped axes, so that the connected slices match
+    # This is only relevant when the boundary width is not equal for all sides.
+    face_0_padded_swapped = face_0.pad(
+        **{
+            x: (boundary_width["Y"][0], 0),
+            y: (boundary_width["X"][1], boundary_width["X"][0]),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+
+    face_1_padded_swapped = face_1.pad(
+        **{
+            x: (boundary_width["Y"][1], boundary_width["Y"][0]),
+            y: (0, boundary_width["X"][1]),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+    return face_0_padded, face_1_padded, face_0_padded_swapped, face_1_padded_swapped
+
+
+def _prepad_right_right_swap_axis(da, boundary_width, fill_value, x="x", y="y"):
+    # Manually put together the data
+    face_0 = da.isel(face=0)
+    face_1 = da.isel(face=1)
+
+    # pad on each side except on the connected side
+    face_0_padded = face_0.pad(
+        **{
+            x: (boundary_width["X"][0], 0),
+            y: (boundary_width["Y"][0], boundary_width["Y"][1]),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+
+    face_1_padded = face_1.pad(
+        **{
+            x: (boundary_width["X"][0], boundary_width["X"][1]),
+            y: (boundary_width["Y"][0], 0),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+
+    # Now pad each face according to the swapped axes, so that the connected slices match
+    # This is only relevant when the boundary width is not equal for all sides.
+    face_0_padded_swapped = face_0.pad(
+        **{
+            x: (boundary_width["Y"][0], 0),
+            y: (boundary_width["X"][0], boundary_width["X"][1]),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+
+    face_1_padded_swapped = face_1.pad(
+        **{
+            x: (boundary_width["Y"][0], boundary_width["Y"][1]),
+            y: (boundary_width["X"][0], 0),
+            "mode": "constant",
+            "constant_values": fill_value,
+        }
+    )
+    return face_0_padded, face_1_padded, face_0_padded_swapped, face_1_padded_swapped
+
+
+@pytest.mark.parametrize("fill_value", [np.nan, 0])
+@pytest.mark.parametrize(
+    "boundary_width",
+    [
+        {"X": (1, 1)},
+        {"X": (1, 2)},
+        {"X": (0, 1)},
+        {
+            "X": (1, 1),
+            "Y": (1, 1),
+        },
+        {
+            "X": (2, 2),
+            "Y": (2, 2),
+        },
+        {
+            "X": (0, 1),
+            "Y": (1, 0),
+        },
+        {
+            "X": (0, 2),
+            "Y": (1, 0),
+        },
+    ],
+)
+class TestPaddingFaceConnection:
```

Comment:
Could you divide this into two classes, one for padding only involving `X` and one for the more complex cases that involve `Y` too?

Reply:
Should we bench that for a later PR as discussed?